static/js/controllers/contacts.js

```diff
@@ -29,6 +30,20 @@ var _ = require('underscore'),
       $scope.selected = null;
       $scope.filters = {};
 
+      var defaultTypeFilter = {};
+      function setDefaultTypeFilter() {
+        return UserSettings().then(function(u) {
+          return u.facility_id && DB().get(u.facility_id).then(function(facility) {
+            var targetType = CONTACT_TYPES[CONTACT_TYPES.indexOf(facility.type) + 1];
```

Comment:
> A more detailed error for the "woe betide" case would be nice.

You're right. I originally believed that it would be OK because a) it would never happen, b) the error would happen consistently if you opened the contacts view and thus would be easy to track down and notice and c) would give you a nice error about the index not being valid for the array. Upon just double-checking that now you don't get an error when you access a value in an array with a bad index (`[1,2,3][-1] === undefined`), because of course JavaScript would do that.

So what will happen is that we'll try to filter by undefined, and who knows what will happen there.

Since you've linked a refactoring ticket above, do you think it's worth changing now or just dealing with in the linked ticket?